whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/LuceneSearchCondition.java

```diff
@@ -0,0 +1,16 @@
+package net.ripe.db.whois.api.autocomplete;
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class LuceneSearchCondition implements Condition {
```

Comment:
Can you implement this class by extending ```ElasticSearchCondition``` and negating the super.matches() response? This eliminates the code duplication and guarantees that both conditions can never be true simultaneously